scripts/generate-chains.ts

```diff
@@ -0,0 +1,64 @@
+import fs from 'fs';
+import path from 'path';
+import chokidar from 'chokidar';
+
+const INPUT_DIR = './chains';
+const OUTPUT_DIR = './src/generated';
+const OUTPUT_FILE = `${OUTPUT_DIR}/chains.ts`;
+
+const BASE_CONTENT = `// ===========================================================================
+// DO NOT EDIT THIS FILE MANUALLY!
+//
+// The contents have been added automatically.
+// See: scripts/combine-chains.ts for more information
+// ===========================================================================
+
+import { Chain } from '../types';
+`;
+
+async function mergeJsonFiles() {
+  const fileNames = await fs.promises.readdir(INPUT_DIR);
```

Comment:
nit: I'd use *Sync methods everywhere for scripts. It fiels a bit wierd that you mix them with promise ones. Doesn't matter in practice, so feel free to leave.